frame/support/procedural/src/construct_runtime/mod.rs

```diff
@@ -471,3 +475,46 @@ fn decl_integrity_test(scrate: &TokenStream2) -> TokenStream2 {
 		}
 	)
 }
+
+fn decl_static_assertions(
+	runtime: &Ident,
+	pallet_decls: &[Pallet],
+	scrate: &TokenStream2,
+) -> TokenStream2 {
+	let error_encoded_size_check = pallet_decls.iter().map(|decl| {
+		let count = COUNTER.with(|counter| counter.borrow_mut().inc());
+		let path = &decl.path;
+		let assert_macro_name = format_ident!("__assert_error_encoded_size_{}", count);
+		let macro_alias = format_ident!("assert_error_encoded_size_{}", count);
+
+		quote! {
+			#scrate::tt_call! {
+				macro = [{ #path::tt_error_token }]
+				frame_support = [{ #scrate }]
+				~~> #macro_alias
+			}
+
+			#[macro_export]
+			#[doc(hidden)]
+			macro_rules! #assert_macro_name {
+				{
+					error = [{ $error:ident }]
+				} => {
+					#scrate::const_assert! {
+						<
+							#path::$error<#runtime> as #scrate::traits::PalletError
+						>::MAX_ENCODED_SIZE <= #scrate::MAX_PALLET_ERROR_ENCODED_SIZE
+					}
```

Comment:
Since 1.57 panic! in const context is valid: https://blog.rust-lang.org/2021/12/02/Rust-1.57.0.html#panic-in-const-contexts

Please use this here, because that enables us to use a proper error message.

Reply:
Weird, I just tried converting this to an `assert!`, but it seems like the code inside of it isn't run during compile time. Here's what I had:

```rust
const _: () = assert!(
	<
		#path::$error<#runtime> as #scrate::traits::PalletError
	>::MAX_ENCODED_SIZE <= #scrate::MAX_PALLET_ERROR_ENCODED_SIZE,
	#assert_message
);
```

I even tried to put a `false` in there to see if it triggers, but no dice.

Reply:
Wtf, so this works:

```rust
const _: [();
	#scrate::MAX_PALLET_ERROR_ENCODED_SIZE - <
		#path::$error<#runtime> as #scrate::traits::PalletError
	>::MAX_ENCODED_SIZE
] = [];
```

but obviously it doesn't produce the required error message so we can't have this...

Reply:
> Weird, I just tried converting this to an `assert!`, but it seems like the code inside of it isn't run during compile time. Here's what I had:
> 
> ```rust
> const _: () = assert!(
> 	<
> 		#path::$error<#runtime> as #scrate::traits::PalletError
> 	>::MAX_ENCODED_SIZE <= #scrate::MAX_PALLET_ERROR_ENCODED_SIZE,
> 	#assert_message
> );
> ```
> 
> I even tried to put a `false` in there to see if it triggers, but no dice.

I just tried this with your branch and it worked for me.